tests/integration/targets/lookup_hashi_vault/lookup_hashi_vault/tasks/jwt_test.yml

```diff
@@ -0,0 +1,48 @@
+- vars:
+    role_id: test-role
+    jwt: '{{ lookup("file", "token.jwt") }}'
+    jwt_invalid: '{{ lookup("file", "token_invalid.jwt") }}'
+  block:
+    - name: 'Fetch secrets using "hashi_vault" lookup'
+      set_fact:
+        secret1: "{{ lookup('community.general.hashi_vault', conn_params ~ 'secret=' ~ vault_kv2_path ~ '/secret1 auth_method=jwt jwt=' ~ jwt ~ ' role_id=' ~ role_id) }}"
+        secret2: "{{ lookup('community.general.hashi_vault', conn_params ~ 'secret=' ~ vault_kv2_path ~ '/secret2 auth_method=jwt jwt=' ~ jwt ~ ' role_id=' ~ role_id) }}"
+
+    - name: 'Check secret values'
+      fail:
+        msg: 'unexpected secret values'
+      when: secret1['value'] != 'foo1' or secret2['value'] != 'foo2'
+
+    - name: 'Failure expected when erroneous credentials are used'
+      vars:
+        secret_wrong_cred: "{{ lookup('community.general.hashi_vault', conn_params ~ 'secret=' ~ vault_kv2_path ~ '/secret2 auth_method=jwt jwt=' ~ jwt_invalid ~ ' role_id=' ~ role_id) }}"
+      debug:
+        msg: 'Failure is expected ({{ secret_wrong_cred }})'
+      register: test_wrong_cred
+      ignore_errors: true
+
+# FIXME: Somehow this test is failing. The role is getting access to secret3 - even if the policy explicitly denies it.
```

Comment:
I'm going to try to pull this down and see if I can figure anything out with the tests locally.

Reply:
And what a job! Great work @briantist! I just wished you did it sooner! 😉 Now ALL tests are green. In addition to following up on the registered hvac issue, I think we should improve our test setup (in another PR). We have to make sure no additional working hours are wasted on a "self-shot" test setup. I understand why Vault is booted in dev-mode, but I think we should have a look at the `-dev-no-store-token` [option](https://www.vaultproject.io/docs/commands/server#dev-options) when starting the Vault server in dev-mode. And include the returned root-token in the vault_cmd variable. At least I think that could work....

Reply:
We're planning to move this plugin to its own collection very soon, which will be a good place to improve testing. Especially since then the plugin won't be limited by the community.general test setup, but you can change the testing environment in whatever way you please.

Reply:
Sorry took a while before I could find the time. Running these tests locally is kind of a pain because it's a custom/`runme.sh` type of test apparently. Just another thing we can hopefully improve by moving the plugin. Good idea on the `-dev-no-store-token` option, thanks for that! I hope you'll consider more contributions in the future :)